File: Project2_Java/src/pl/pg/oop/project2/organisms/animals/Antelope.java
package pl.pg.oop.project2.organisms.animals;

import pl.pg.oop.project2.organisms.Animal;
import pl.pg.oop.project2.organisms.OrganismsEnum;
import pl.pg.oop.project2.organisms.World;

import java.awt.*;

public class Antelope extends Animal {
    private final static int DEFAULT_INITIATIVE = 4;
    private final static int DEFAULT_STRENGTH = 4;
    private final static int DEFAULT_STEPS = 2;

    public Antelope(World World_, Point Position_) {
        super(World_, Position_);
        TypeOfOrganism = OrganismsEnum.ANTELOPE;
        initiative = DEFAULT_INITIATIVE;
        strength = DEFAULT_STRENGTH;
        step = DEFAULT_STEPS;
    }

    public Antelope(World world, Point position, int strength) {
        super(world, position, strength);
        this.TypeOfOrganism = OrganismsEnum.ANTELOPE;
        this.initiative = DEFAULT_INITIATIVE;
        step = DEFAULT_STEPS;
    }

    @Override
    public String toString() {
        return "Antelope";
    }
}

File: Project3_Python/Animals/Antelope.py
from Animal import Animal
from Organism import OrganismsEnum


class Antelope(Animal):
    __DEFAULT_INITIATIVE = 4
    __DEFAULT_STRENGTH = 4
    __DEFAULT_STEPS = 2

    def __init__(self, world, position, strength=__DEFAULT_STRENGTH):
        super().__init__(world, position, strength)
        self.type_of_organism = OrganismsEnum.Antelope
        self._initiative = Antelope.__DEFAULT_INITIATIVE
        self._step = Antelope.__DEFAULT_STEPS

    def __str__(self):
        return "Antelope"
